core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -85,15 +95,38 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options) throws IOExcepti
     mUnderStorageType = options.getUnderStorageType();
     mContext = FileSystemContext.INSTANCE;
     mPreviousBlockOutStreams = new LinkedList<BufferedBlockOutStream>();
+    mUfsDelegation = ClientContext.getConf().getBoolean(Constants.USER_UFS_OPERATION_DELEGATION);
     if (mUnderStorageType.isSyncPersist()) {
-      updateUfsPath();
-      String tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
-      UnderFileSystem ufs = UnderFileSystem.get(tmpPath, ClientContext.getConf());
-      // TODO(jiri): Implement collection of temporary files left behind by dead clients.
-      mUnderStorageOutputStream = ufs.create(tmpPath, (int) mBlockSize);
+      if (mUfsDelegation) {
+        updateUfsPath();
+        mFileSystemWorkerClient = mContext.acquireWorkerClient();
+        try {
+          mUfsFileId =
+              mFileSystemWorkerClient.createUfsFile(new AlluxioURI(mUfsPath),
+                  CreateUfsFileOptions.defaults());
+        } catch (AlluxioException e) {
+          mFileSystemWorkerClient.close();
+          throw new IOException(e);
+        }
+        mUnderStorageOutputStream =
+            new UnderFileSystemFileOutStream(mFileSystemWorkerClient.getWorkerDataServerAddress(),
+                mUfsFileId);
+      } else {
+        updateUfsPath();
+        String tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
+        UnderFileSystem ufs = UnderFileSystem.get(tmpPath, ClientContext.getConf());
```

Comment:
I think we should call `ufs.connectFromWorker()` here, so that (with delegation) Worker will be able to access secure HDFS.